confmap/confmap.go

```diff
@@ -285,7 +286,8 @@ func unmarshalerEmbeddedStructsHookFunc() mapstructure.DecodeHookFuncValue {
 		}
 		for i := 0; i < to.Type().NumField(); i++ {
 			// embedded structs passed in via `squash` cannot be pointers. We just check if they are structs:
-			if to.Type().Field(i).IsExported() && to.Type().Field(i).Anonymous {
+			f := to.Type().Field(i)
+			if f.IsExported() && slices.Contains(strings.Split(f.Tag.Get("mapstructure"), ","), "squash") {
```

Comment:
If we remove `Anonymous` from the condition, where is the `Ummarshal` called on the embedded structs without the `squash` tag? I thought this hook is expected to handle embedded structs regardless of field tags.

Reply:
Ah, right, the `doc_test.go` should do the trick